runtime/src/snapshot_utils.rs

```diff
@@ -1596,6 +1599,25 @@ pub fn verify_snapshot_archive<P, Q, R>(
 
     // Check snapshots are the same
     let unpacked_snapshots = unpack_dir.join("snapshots");
+    if let Some(slot) = slot {
+        // file contents may be different, but deserialized structs should be equal
+        let mut p1 = PathBuf::from(snapshots_to_verify.clone());
+        p1 = p1.join(slot.to_string());
+        p1 = p1.join(slot.to_string());
+        let mut p2 = unpacked_snapshots.clone();
+        p2 = p2.join(slot.to_string());
+        p2 = p2.join(slot.to_string());
+
+        assert!(crate::serde_snapshot::compare_two_serialized_banks(
+            SerdeStyle::Newer,
+            p1.clone(),
+            p2.clone()
+        )
```

Comment:
Then this can become?
```suggestion
        assert!(crate::serde_snapshot::compare_two_serialized_banks(
            SerdeStyle::Newer,
            &p1,
            &p2
        )
```